### ISSUE TITLE
Can't clone branch ref name with hash/number sign (#)

### ISSUE THREAD

**Issue by jkblume** [go-git/go-git#309]
I am cloning a repository in single branch mode with a branch name that contains a hash/number sign (#) into name. And get an error.

Steps to reproduce (created a public github repository for that):

```
_, err := git.PlainClone("/tmp/repo", false, &git.CloneOptions{
    URL:           "git@github.com:jkblume/gogit-hashtag-bug.git",
    ReferenceName: plumbing.ReferenceName("refs/heads/branch#name"),
    SingleBranch:  true,
})

fmt.Println(err.Error())
```

The command fails with error message: 'malformed refspec, separators are wrong' in file config/refspec.go line 35.

**Comment by my-gitter-im** [comment]
I've raised a PR with a possible fix for this but I haven't tested it.
[PR #353](https://github.com/go-git/go-git/pull/353)
Can someone add some reviewers? I don't seem to be able to.

**Comment by dowy** [comment]
I've fixed this issue in [PR#354](https://github.com/go-git/go-git/pull/354)
Please review

**Comment by AgnesToulet** [comment]
Hi! I have the same issue. Is there any work-around to make this work? (I'm trying to escape the hash sign but can't find a syntax that works).

**Comment by mbirbeck** [comment]
I'm also hitting this issue. I tried adding the changes in PR https://github.com/go-git/go-git/pull/354 as suggested above and it fixed the issue. @mcuadros or @erizocosmico do you think we could get this PR merged to master?

**Comment by mbirbeck** [comment]
@mcuadros or @erizocosmico the fix in PR #354 is changing a lot more things than just fixing the issue with the # character in branch names, so I have submitted PR #611 as a targeted fix for just this particular issue.

**Comment by dowy** [comment]
> @mcuadros or @erizocosmico the fix in PR #354 is changing a lot more things than just fixing the issue with the # character in branch names, so I have submitted PR #611 as a targeted fix for just this particular issue.

That's true, but those other changes will make the code correctly write other kinds of config-file entries -- and #354 has tests!
